dom.go

```diff
@@ -1946,120 +1992,196 @@ func (e *HTMLCanvasElement) GetContext(param string) *js.Object {
 	return e.Call("getContext", param)
 }
 
-// Colors, Styles, and Shadows
+// Drawing Rectangles
 
-func (ctx *CanvasRenderingContext2D) CreateLinearGradient(x0, y0, x1, y1 int) {
-	ctx.Call("createLinearGradient", x0, y0, x1, y1)
+func (ctx *CanvasRenderingContext2D) ClearRect(x, y, width, height float64) {
+	ctx.Call("clearRect", x, y, width, height)
 }
 
-// Rectangles
+func (ctx *CanvasRenderingContext2D) FillRect(x, y, width, height float64) {
+	ctx.Call("fillRect", x, y, width, height)
+}
 
-func (ctx *CanvasRenderingContext2D) Rect(x, y, width, height int) {
-	ctx.Call("rect", x, y, width, height)
+func (ctx *CanvasRenderingContext2D) StrokeRect(x, y, width, height float64) {
+	ctx.Call("strokeRect", x, y, width, height)
 }
 
-func (ctx *CanvasRenderingContext2D) FillRect(x, y, width, height int) {
-	ctx.Call("fillRect", x, y, width, height)
+// Drawing Text
+
+func (ctx *CanvasRenderingContext2D) FillText(text string, x, y, maxWidth float64) {
+	if maxWidth == -1 {
+		ctx.Call("fillText", text, x, y)
+		return
+	}
+
+	ctx.Call("fillText", text, x, y, maxWidth)
 }
 
-func (ctx *CanvasRenderingContext2D) StrokeRect(x, y, width, height int) {
-	ctx.Call("strokeRect", x, y, width, height)
+func (ctx *CanvasRenderingContext2D) StrokeText(text string, x, y, maxWidth float64) {
+	if maxWidth == -1 {
+		ctx.Call("strokeText", text, x, y)
+		return
+	}
+
+	ctx.Call("strokeText", text, x, y, maxWidth)
+}
+func (ctx *CanvasRenderingContext2D) MeasureText(text string) *TextMetrics {
+	textMetrics := ctx.Call("measureText", text)
+	return &TextMetrics{Object: textMetrics}
 }
 
-func (ctx *CanvasRenderingContext2D) ClearRect(x, y, width, height int) {
-	ctx.Call("clearRect", x, y, width, height)
+// Line styles
+
+func (ctx *CanvasRenderingContext2D) GetLineDash() interface{} {
+	return ctx.Call("getLineDash")
 }
 
-// Paths
+//
+func (ctx *CanvasRenderingContext2D) SetLineDash(arr interface{}) {
```

Comment:
What is `arr`? This is pretty unclear.

I see at https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/setLineDash they're calling the parameter `segments`.

I know this library doesn't set a good precedent for documenting exported identifiers, but it's never too late to start, and this could be a good case where including documentation would be very helpful. How about:

```Go
// SetLineDash sets the line dash pattern used when stroking lines.
// segments is an array of numbers which specify distances to alternately
// draw a line and a gap (in coordinate space units). If the number of
// elements in the array is odd, the elements of the array get copied
// and concatenated. For example, [5, 15, 25] will become [5, 15, 25, 5, 15, 25].
// If the array is empty, the line dash list is cleared and line strokes return to being solid.
func (ctx *CanvasRenderingContext2D) SetLineDash(segments []int) {
}
```

It also looks like we can use `[]int` instead of `interface{}` as the type. Do you agree, or am I missing anything?

Reply:
If we're sticking with `interface{}` argument here, we definitely need documentation to explain how to use it. See first comment for a starting point.

Reply:
Ok so i went with []float64, it seems just easier to use anyway, as the function does work with both value types, interface{} seems an exageration as we can easily fit int values in floats.